### PR TITLE
Add credentials_endpoint to readme

### PR DESCRIPTION
**Pull request by PettitWesley**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +28,16 @@ Run `make` to build `./bin/cloudwatch.so`. Then use with Fluent Bit:
 * `role_arn`: ARN of an IAM role to assume (for cross account access).
 * `auto_create_group`: Automatically create the log group. Valid values are "true" or "false" (case insensitive). Defaults to false.
 * `endpoint`: Specify a custom endpoint for the CloudWatch Logs API.
+* `credentials_endpoint`: Specify a custom HTTP endpoint to pull credentials from. The HTTP response body should look like the following:
+```
+{
+    "AccessKeyId": "ACCESS_KEY_ID",
+    "Expiration": "EXPIRATION_DATE",
+    "RoleArn": "TASK_ROLE_ARN",
```
Comment (kajeeth) [inline README.md]: Lookslike SDK does not support RoleArn in endpointcreds. Shall we remove it?

https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/endpointcreds/

**Comment by guangchuan** [comment]
@PettitWesley 
Hello 
I'm trying to use credentials_endpoint, how do I use it?

I am testing as follows.

`$ docker run --rm -p 127.0.0.1:24224:24224 amazon/aws-for-fluent-bit:latest /fluent-bit/bin/fluent-bit -e /fluent-bit/cloudwatch.so -i cpu -o cloudwatch -p "region=myregion" -p "log_group_name=myloggroup" -p "log_stream_name=fluent-bit" -p "credentials_endpoint=myapi-gateway.amazonaws.com/prd"`

And the api-gageway(API Gateway->Lambda) is returning the json as per the README.
However, I get an error.

`panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x7ff7fbed1dd0]`

What format do you want for the credentials_endpoint response?
May I tell you how to use credentials_endpoint?
Thank you.

**Comment by PettitWesley** [comment]
@guangchuan `credentials_endpoint` is expected to be a local endpoint, something like the endpoint [ECS adds for IAM Roles for Tasks](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html). I am doubtful that the code will work with an external endpoint. At that link, you can see the expected JSON response for the endpoint:

```
{
    "AccessKeyId": "ACCESS_KEY_ID",
    "Expiration": "EXPIRATION_DATE",
    "RoleArn": "TASK_ROLE_ARN",
    "SecretAccessKey": "SECRET_ACCESS_KEY",
    "Token": "SECURITY_TOKEN_STRING"
}
```

**Comment by PettitWesley** [comment]
@guangchuan Why are you trying to vend credentials from an API Gateway endpoint? Sorry, but that sounds like a terrible idea from a security point of view. This option was added for some special use cases which we have; in general, as an AWS customer it is best to use one of the standard credential mechanisms (like local aws profile, environment variables, etc).

What is your use case?

**Comment by guangchuan** [comment]
@PettitWesley 
Thank you for your reply.

As a use case, I'm trying to send logs from fluent-bit on-premises to CloudWatch Logs, but I'm trying to rotate without fixing access_key and secretaccess_key.

I understand that this option is for ECS only. Thank you very much.

**Pull request merged**